listenbrainz_spark/stats/user/utils.py

```diff
@@ -1,51 +1,31 @@
+from datetime import datetime
 import time
 
 from collections import defaultdict
 from listenbrainz_spark.stats import run_query
+from listenbrainz_spark.utils import get_listens
+from listenbrainz_spark.stats import adjust_months
+from listenbrainz_spark.path import LISTENBRAINZ_DATA_DIRECTORY
 
-def get_artists(table):
-    """
-    Get artist information (artist_name, artist_msid etc) for every user
-    ordered by listen count (number of times a user has listened to tracks
-    which belong to a particular artist).
 
-    Args:
-        table: name of the temporary table.
+def get_latest_listen_ts():
+    now = datetime.now()
+    while True:
+        try:
+            df = get_listens(now, now, LISTENBRAINZ_DATA_DIRECTORY)
+            break
+        except:
```

Comment:
this `except` catches ALL exceptions, that's bad, because it might catch exceptions that you don't want it to catch, I'd only catch the exception that you care about here.